### ISSUE TITLE
Forward declarations are no longer allowed in type-erased continuables

### ISSUE THREAD

**Issue by Rogiel** [Naios/fluent.cpp#11]
Hello @Naios 

Somewhere after 4d58e3b `Ts` in cti::continuables<Ts...> can no longer have types with a forward declaration when declaring the continuable as a return type in a function call.

```c++
struct forward_decl_type;
cti::continuable<forward_decl_type> fails_with_forward_decls();
```

This simple statement compiles fine with `4d58e3b` but fails in the current `master` branch.

-----

### Commit Hash

Working fine in `4d58e3b`, not working in `master`.

### Expected Behavior

A header containing the declaration above should compile fine (as it did in `4d58e3b`) without the need to fully define `forward_decl_type`. The compilation can (and should) still fail when calling `fails_with_forward_decls` without fully defining `forward_decl_type`.

### Actual Behavior

The declaration of `fails_with_forward_decls` returning a `cti::continuable<forward_decl_type>` when `forward_decl_type` is a forward declaration fails to compile.

### Your Environment

- OS: Windows 10
- Compiler and version: Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27025.1 for x64

**Comment by Rogiel** [comment]
The same problem occurs with the latest `AppleClang` compiler.

**Comment by Naios** [comment]
Thank you for your bug report, I'll take a look into this.

**Comment by Naios** [comment]
The issue is fixed in c066940, note that function2 was updated to v4.0.0 and requires an update too.
The bug was that the signature of the type erasure wrapper was dependent on the size of the forward declared type, a unit test was added to prevent this bug in the future.
